File: solutions/Java/binary_tree_level_order_traversal_bfs.java
import java.util.*;

public class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<>();
        if (root == null) return result;
        Queue<TreeNode> q = new LinkedList<>(); q.offer(root);
        while (!q.isEmpty()) {
            int sz = q.size(); List<Integer> level = new ArrayList<>();
            for (int i = 0; i < sz; i++) {
                TreeNode n = q.poll();
                level.add(n.val);
                if (n.left != null) q.offer(n.left);
                if (n.right != null) q.offer(n.right);
            }
            result.add(level);
        }
        return result;
    }
    
    public static class TreeNode {
        int val; TreeNode left, right;
        TreeNode(int x) { val = x; }
    }
}

File: solutions/JavaScript/binary_tree_level_order_traversal_bfs.js
function TreeNode(val, left = null, right = null) { this.val = val; this.left = left; this.right = right; }

function levelOrder(root) {
    const result = []; if (!root) return result;
    const queue = [root];
    while (queue.length) {
        const sz = queue.length, level = [];
        for (let i = 0; i < sz; i++) {
            const n = queue.shift();
            level.push(n.val);
            if (n.left) queue.push(n.left);
            if (n.right) queue.push(n.right);
        }
        result.push(level);
    }
    return result;
}

module.exports = levelOrder;
